File: app/src/main/java/com/example/gemini_ai_chat/Message.java

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "messages")
public class Message {
    @PrimaryKey(autoGenerate = true)
    public int id;

    public String text;
    public boolean isUser;
    public long timestamp;

    public Message(String text, boolean isUser, long timestamp) {
        this.text = text;
        this.isUser = isUser;
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/example/gemini_ai_chat/Message.kt

package com.example.gemini_ai_chat

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "messages")
data class Message(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val text: String,
    val isUser: Boolean,
    val timestamp: Long
)
